eng/pipelines/common/evaluate-paths-job.yml

```diff
@@ -0,0 +1,53 @@
+### Job used to evaluate changed paths on a pull request to emit variables to condition jobs based on paths.
+
+parameters:
+  # Object containing subset include and exclude paths in an array form.
+  # Scenarios:
+  #  1. exclude paths are specified
+  #     Will include all paths except the ones in the exclude list.
+  #  2. include paths are specified
+  #     Will only include paths specified in the list.
+  #  3. exclude + include:
+  #     1st we evaluate changes for all paths except ones in excluded list. If we can't find
+  #     any applicable changes like that, then we evaluate changes for incldued paths
+  #     if any of these two finds changes, then a variable will be set to true.
+  #  In order to consume this variable you need to reference it via: $[ dependencies.evaluate_paths.outputs['SetPathVars_<subset>.containschange'] ]
+  #
+  #  Array form example
+  #  paths:
+  #  - subset: coreclr
+  #    include:
+  #    - src/libraries/System.Private.CoreLib/*
+  #    exclude:
+  #    - src/libraries/*
+  #
+  #  This example will include ALL path changes under src/libraries/ except the ones under src/libraries/*!System.Private.CoreLib/*
+  paths: []
+
+
+jobs:
+- ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
should this instead check on the dependOnEvaluatePaths variable?

Reply:
This template doesn't go through that code path. There is no variable, it is a parameter. I can think of another way to do this for the job expansion in a more shareable way 😄

Reply:
You mean can't, right? :D

Reply:
No I meant can. I will think of it, I'm sure we can make this more intuitive by sharing a value somehow 😄 .

Reply:
Awesome